inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -827,3 +829,87 @@ def extract_authors_from_pdf(obj, eng):
+    multi_search = MultiSearch(index="records-experiments", using=current_search_client)
+    for collaboration in collaborations:
+        full_collaboration_string = collaboration.get('value', '')
+        normalized_collaboration_string = normalize_collaboration_name(full_collaboration_string)
+        if collaboration.get('record'):
+            # Add dummy search so multisearch will stay in sync with collaborations
+            multi_search = multi_search.add(Search().query().source(False))
+            multi_search = multi_search.add(Search().query().source(False))
+            continue
+        name_search, subgroup_search = build_collaboration_search(normalized_collaboration_string)
+        multi_search = multi_search.add(name_search)
+        multi_search = multi_search.add(subgroup_search)
+
+    return multi_search
+
+
+def build_collaboration_search(normalized_collaboration_string):
+    name_search = Q("term", normalized_name_variants={"value": normalized_collaboration_string})
+    subgroup_search = Q("term", normalized_subgroups={"value": normalized_collaboration_string})
+    source = ['collaboration', 'self', 'legacy_name', 'control_number']
+    filters_ = Q("exists", field="collaboration")
+    return Search().query(name_search).filter(filters_).source(source),\
+        Search().query(subgroup_search).filter(filters_).source(source),
+
+
+def find_subgroup(subgroup, experiment):
+    clean_special_characters = re.compile(r"[^\w\d_]", re.UNICODE)
+    normalized_subgroup = normalize_collaboration_name(clean_special_characters.sub(' ', subgroup.lower()))
+    subgroups = experiment.collaboration.subgroup_names
+    normalized_subgroups = [normalize_collaboration_name(clean_special_characters.sub(' ', element.lower())) for element in subgroups]
+    for idx, subgroup_from_list in enumerate(normalized_subgroups):
+        if subgroup_from_list == normalized_subgroup:
+            return subgroups[idx]
+    raise CannotFindProperSubgroup(experiment['control_number'], subgroup)
+
+
+@with_debug_logging
+def normalize_collaborations(obj, eng):
+    collaborations = obj.data.get('collaborations', [])
+    if not isinstance(collaborations, list) or len(collaborations) < 1:
+        LOGGER.exception("Metadata are malformed for record %s. Collaborations key is not a list.", obj.id)
+        return obj
+    multi_search = prepare_collaboration_multi_search(collaborations)
+    try:
+        search_responses = multi_search.execute()
+    except ValueError as er:
+        # Ignore normalization on ES errors
+        LOGGER.exception("Cannot perform collaborations normalization due to ES error: %s", er)
+        return obj
+    if not len(search_responses) == len(collaborations) * 2:
+        LOGGER.exception("Results count does not match collaborations count: %s : %s. wf id: %s", len(collaborations), len(search_responses), obj.id)
+        return obj
+    for idx, collaboration in enumerate(collaborations):
```

Comment:
I think it's easier to transform `search_responses` into an iterator of pairs as in https://stackoverflow.com/a/23286299, so you don't need to do arithmetic, but can simply do
```suggestion
    for collaboration, (collaboration_response, subgroup_response) in zip(collaborations, grouped_search_responses):
```

Reply:
I still need index of current collaboration to update it on the record collaboration list :)

Reply:
no you don't, you can mutate the `dict` instead :)

Reply:
Ah, right ;)